### PR TITLE
Pluggable backoff

### PR DESCRIPTION
**Pull request by acd**

### CODE REVIEW

**Comment by acd** [comment]
I chose this implementation to maintain backward compatibility for existing users of the library.

**Comment by pin** [comment]
LGTM

Seems like a reasonable way to allow to replace default jitter with something else.

Looks like merge of https://github.com/pin/tftp/pull/23 somehow conflicted, please rebase whenever you have a chance!

**Comment by acd** [comment]
Rebased, branch should merge cleanly now.

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ func readHandler(filename string, rf io.ReaderFrom) error {
 }
 
 // writeHandler is called when client starts file upload to server
-func writeHanlder(filename string, wt io.WriterTo) error {
+func writeHandler(filename string, wt io.WriterTo) error {
```
Comment (pin) [inline README.md]: 👍

**Comment by pin** [comment]
Looks great, merging.

**Pull request merged**